cartridges/custom_cartridge_1/cartridge/controllers/LoggedOrGuestUser.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+/**
+ * @namespace LoggedOrGuestUser
+ */
+
+var server = require("server");
+var cache = require("*/cartridge/scripts/middleware/cache");
+var consentTracking = require("*/cartridge/scripts/middleware/consentTracking");
+var pageMetaData = require("*/cartridge/scripts/middleware/pageMetaData");
+/**
+ * 
+/**
+ * @name Base/LoggedOrGuestUser-Show
+ * @function
+ * @memberof LoggedOrGuestUser
+ * @param {middleware} - consentTracking.consent
+ * @param {middleware} - cache.applyDefaultCache
+ * @param {category} - non-sensitive
+ * @param {renders} - isml
+ * @param {serverfunction} - get
+ */
+server.get(
+    "Show",
+    consentTracking.consent,
+    cache.applyDefaultCache,
+    function (req, res, next) {
+        var Site = require("dw/system/Site");
+        var PageMgr = require("dw/experience/PageMgr");
+        var pageMetaHelper = require("*/cartridge/scripts/helpers/pageMetaHelper");
+        var ContentMgr = require('dw/content/ContentMgr');
+
+        pageMetaHelper.setPageMetaTags(req.pageMetaData, Site.current);
+
+        var loggedInAsset = ContentMgr.getContent('Logged');
+        var guestAsset = ContentMgr.getContent('Guest');
+        var loggedInAssetBody = loggedInAsset.custom.body;
+        var guestAssetBody = guestAsset.custom.body;
```

Comment:
This can be optimized by loading the correct asset in the if clauses below. Now you load both, but only one will be used.

Reply:
It is optimized.